File: VB-Examples/18_PageSetup/SetOtherPrintingOptions.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet

Namespace SetOtherPrintingOptions
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load an existing Excel file from the specified path
            workbook.LoadFromFile("../../../../../../Data/Template_Xls_1.xlsx")

            ' Retrieve the first worksheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Get the PageSetup object for the worksheet
            Dim pageSetup As PageSetup = sheet.PageSetup

            ' Set the property to print gridlines on the worksheet
            pageSetup.IsPrintGridlines = True

            ' Set the property to print row and column headings on the worksheet
            pageSetup.IsPrintHeadings = True

            ' Set the property to print in black and white
            pageSetup.BlackAndWhite = True

            ' Set the property to print comments in place
            pageSetup.PrintComments = PrintCommentType.InPlace

            ' Set the property to print in draft mode
            pageSetup.Draft = True

            ' Set the property to handle errors as "Not Available" during printing
            pageSetup.PrintErrors = PrintErrorsType.NA

            ' Specify the file name for the resulting Excel file
            Dim result As String = "Result-SetOtherPrintOptionsOfXlsFile.xlsx"

            ' Save the modified workbook to the specified file in Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/18_PageSetup/SetOtherPrintingOptions.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;

namespace SetOtherPrintingOptions
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a workbook.
            Workbook workbook = new Workbook();

            // Load the file from disk.
            workbook.LoadFromFile(@"../../../../../../Data/Template_Xls_1.xlsx");

            // Get the first worksheet.
            Worksheet sheet = workbook.Worksheets[0];

            // Get the reference of the PageSetup of the worksheet.
            PageSetup pageSetup = sheet.PageSetup;

            // Allow to print gridlines.
            pageSetup.IsPrintGridlines = true;

            // Allow to print row/column headings.
            pageSetup.IsPrintHeadings = true;

            // Allow to print worksheet in black & white mode.
            pageSetup.BlackAndWhite = true;

            // Allow to print comments as displayed on worksheet.
            pageSetup.PrintComments = PrintCommentType.InPlace;

            // Allow to print worksheet with draft quality.
            pageSetup.Draft = true;

            // Allow to print cell errors as N/A.
            pageSetup.PrintErrors = PrintErrorsType.NA;

            // Specify the output file name for the result
            String result = "Result-SetOtherPrintOptionsOfXlsFile.xlsx";

            // Save the modified workbook to the specified file using Excel 2013 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}
